ethereum/contracts/interfaces/relayer/TypedUnits.sol

```diff
@@ -0,0 +1,178 @@
+// SPDX-License-Identifier: Apache 2
+
+pragma solidity ^0.8.19;
+
+type WeiPrice is uint256;
+
+type GasPrice is uint256;
+
+type Gas is uint256;
+
+type Dollar is uint256;
+
+type Wei is uint256;
+
+using {
+    addWei as +,
+    subWei as -,
+    lteWei as <=,
+    ltWei as <,
+    gtWei as >,
+    eqWei as ==,
+    neqWei as !=
+} for Wei global;
+using {ltGas as <, subGas as -} for Gas global;
+
+using WeiLib for Wei;
+using GasLib for Gas;
+using DollarLib for Dollar;
+using WeiPriceLib for WeiPrice;
+using GasPriceLib for GasPrice;
+
+function ltWei(Wei a, Wei b) pure returns (bool) {
+    return Wei.unwrap(a) < Wei.unwrap(b);
+}
+
+function eqWei(Wei a, Wei b) pure returns (bool) {
+    return Wei.unwrap(a) == Wei.unwrap(b);
+}
+
+function gtWei(Wei a, Wei b) pure returns (bool) {
+    return Wei.unwrap(a) > Wei.unwrap(b);
+}
+
+function lteWei(Wei a, Wei b) pure returns (bool) {
+    return Wei.unwrap(a) <= Wei.unwrap(b);
+}
+
+function subWei(Wei a, Wei b) pure returns (Wei) {
+    return Wei.wrap(Wei.unwrap(a) - Wei.unwrap(b));
+}
+
+function addWei(Wei a, Wei b) pure returns (Wei) {
+    return Wei.wrap(Wei.unwrap(a) + Wei.unwrap(b));
+}
+
+function neqWei(Wei a, Wei b) pure returns (bool) {
+    return Wei.unwrap(a) != Wei.unwrap(b);
+}
+
+function ltGas(Gas a, Gas b) pure returns (bool) {
+    return Gas.unwrap(a) < Gas.unwrap(b);
+}
+
+function subGas(Gas a, Gas b) pure returns (Gas) {
+    return Gas.wrap(Gas.unwrap(a) - Gas.unwrap(b));
+}
+
+library WeiLib {
+    using {toDollars, toGas, convertAsset, min, scale, unwrap} for Wei;
+
+    Wei constant ZERO = Wei.wrap(0);
+
+    function min(Wei x, Wei maxVal) internal pure returns (Wei) {
+        return x > maxVal ? maxVal : x;
+    }
+
+    function toDollars(Wei w, WeiPrice price) internal pure returns (Dollar) {
+        return Dollar.wrap(Wei.unwrap(w) * WeiPrice.unwrap(price));
+    }
+
+    function toGas(Wei w, GasPrice price) internal pure returns (Gas) {
+        return Gas.wrap(Wei.unwrap(w) / GasPrice.unwrap(price));
+    }
+
+    function scale(Wei w, Gas num, Gas denom) internal pure returns (Wei) {
+        return Wei.wrap(Wei.unwrap(w) * Gas.unwrap(num) / Gas.unwrap(denom));
+    }
+
+    function unwrap(Wei w) internal pure returns (uint256) {
```

Comment:
Agreed, but isn't that true with all the methods attached to these units? Personally the visual clutter is too much for me without the postfix and the impl can't be changed out from under us since it's compiled into the contract so I lean towards keeping the suffix. Lmk if you feel strongly the other way or disagree with my reasoning

Reply:
No strong feelings, the only difference is that `wrap` and `unwrap` are the only methods that come with the type itself i.e. `Wei.wrap` and `Wei.unwrap` have a meaning even without `using` WeiLib for `Wei` so it creates a possible ambiguity after the `using` directive (let's say if we were to introduce value checking in our manual version of `unwrap`).

Does that make sense/do you see what I'm saying?